typing/includemod.ml

```diff
@@ -451,21 +451,35 @@ and try_modtypes ~in_eq ~loc env ~mark subst mty1 mty2 orig_shape =
         functor_param ~in_eq ~loc env ~mark:(negate_mark mark)
           subst param1 param2
       in
-      (* Using a fresh variable with a placeholder uid here is fine: users will
-         never try to jump to the definition of that variable.
-         If they try to jump to the parameter from inside the functor, they will
-         use the variable shape that is stored in the local environment.  *)
-      let var, shape_var = Shape.fresh_var Uid.internal_not_actually_unique in
-      let cc_res =
-        let res_shape = Shape.app orig_shape ~arg:shape_var in
-        modtypes ~in_eq ~loc env ~mark subst res1 res2 res_shape
+      let var, res_shape =
+        match orig_shape.Shape.desc with
+        | Abs (var, res_shape) -> var, res_shape
+        | _ ->
```

Comment:
As @Octachron already pointed out, this would be nicer as a utility function in the shape module:
```ocaml
val decompose_abs: t -> var * t
```

(I'm the one who originally wrote this, but since you've now taken ownership \o/)

Reply:
done in a new commit

(I didn't make the type `Shape.t` (or `Shape.desc`?) abstract, I'm not sure that the code improves when those AST types are more abstract. I'll let other decide what is best and send a PR if they care.)